src/main/java/com/arpnetworking/metrics/common/sources/CollectdHttpSourceV1.java

```diff
@@ -0,0 +1,251 @@
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.exception.ConstraintsViolatedException;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Processes HTTP posts from Collectd, extracts data and emits metrics.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class CollectdHttpSourceV1 extends ActorSource {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Props createProps() {
+        return Actor.props(this);
+    }
+
+    /**
+     * Protected constructor.
+     *
+     * @param builder Instance of <code>Builder</code>.
+     */
+    protected CollectdHttpSourceV1(final Builder builder) {
+        super(builder);
+        _parser = builder._parser;
+    }
+
+    private final Parser<List<Record>, com.arpnetworking.metrics.mad.model.HttpRequest> _parser;
+
+    /**
+     * Internal actor to process requests.
+     */
+    /* package private */ static final class Actor extends UntypedActor {
+        /**
+         * Creates a {@link Props} for this actor.
+         *
+         * @param source The {@link CollectdHttpSourceV1} to send notifications through.
+         * @return A new {@link Props}
+         */
+        /* package private */ static Props props(final CollectdHttpSourceV1 source) {
+            return Props.create(Actor.class, source);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onReceive(final Object message) throws Exception {
+            if (message instanceof RequestReply) {
+                final RequestReply requestReply = (RequestReply) message;
+                // TODO(barp): Fix the ugly HttpRequest cast here due to java vs scala dsl
+                Source.single((HttpRequest) requestReply.getRequest())
+                        .via(_processGraph)
+                        .toMat(_sink, Keep.right())
+                        .run(_materializer)
+                        .whenComplete((done, err) -> {
+                            final CompletableFuture<HttpResponse> responseFuture = requestReply.getResponse();
+                            if (err == null) {
+                                responseFuture.complete(HttpResponse.create().withStatus(200));
+                            } else {
+                                BAD_REQUEST_LOGGER.warn()
+                                        .setMessage("Error handling collectd post")
+                                        .setThrowable(err)
+                                        .log();
+                                if ((err instanceof IOException && err.getCause() instanceof ParsingException)
```

Comment:
Should we move the CVE catch into the parser implementation? I imagine this is from some additional builder invocation inside the parser.